sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
-        self.add_cmd_output("kubectl version")
-        self.add_cmd_output("kubectl get -o json pods")
-        self.add_cmd_output("kubectl get -o json nodes")
-        self.add_cmd_output("kubectl get -o json services")
-        self.add_cmd_output("kubectl get -o json replicationController")
-        self.add_cmd_output("kubectl get -o json events")
-        self.add_cmd_output("journalctl -u kubelet")
-        self.add_cmd_output("journalctl -u kube-apiserver")
-        self.add_cmd_output("journalctl -u kube-controller-manager")
-        self.add_cmd_output("journalctl -u kube-scheduler")
-        self.add_cmd_output("journalctl -u kube-proxy")
-
-        if self.get_option('podslog'):
-            result = self.get_command_output("kubectl get pods")
-            if result['status'] == 0:
-                for line in result['output'].splitlines()[1:]:
-                    pod_name = line.split(" ")[0]
-                    self.add_cmd_output([
-                        "{0} log {1}".format("kubectl", pod_name)
-                    ])
+        svcs = [
+            'kubelet', 'kube-apiserver', 'kube-proxy', 'kube-scheduler',
+            'kube-controller-manager'
+        ]
+
+        for svc in svcs:
+            self.add_cmd_output("journalctl -u {}".format(svc))
+
+        # We can only grab kubectl output from the master
+        if self.check_is_master():
+            kube_bin = "kubectl "
+            if path.exists('/etc/origin/master/admin.kubeconfig'):
+                kube_bin += "--config=/etc/origin/master/admin.kubeconfig"
+
+            kube_get = "get -o json "
+            if self.get_option('all'):
+                kube_get += "--all-namespaces=true"
+
+            for cmd in ['version', 'config view']:
+                self.add_cmd_output("{} {}".format(kube_bin, cmd))
+
+            resources = [
+                'events', 'limitrange', 'pods', 'pvc', 'rc',
+                'resourcequota', 'services'
+            ]
+
+            for res in resources:
+                self.add_cmd_output(
+                    "{} {} {}".format(
+                        kube_bin,
+                        kube_get,
+                        res
+                    )
+                )
+            resources.remove('events')
+            # nodes and pvs are not namespaced, must pull separately
+            self.add_cmd_output([
+                "{} get -o json nodes".format(kube_bin),
+                "{} get -o json pv".format(kube_bin)
+                ]
+            )
+
+            if self.get_option('describe'):
+                resources = resources + ['nodes', 'namespaces']
+                for res in resources:
+                    r = self.get_command_output(
+                        "{} get {}".format(
+                            kube_bin,
+                            res
+                        )
+                    )
+                    if r['status'] == 0:
+                        k_list = [k.split()[0] for k in
```

Comment:
Generally this feels like it's over-nested: perhaps this should be broken up into a separate function.